packages/imperative/src/censor/src/Censor.ts

```diff
@@ -0,0 +1,340 @@
+                if (prop.optionDefinition) {
+                    secureOptions.add(prop.optionDefinition.name);
+                    for (const alias of prop.optionDefinition.aliases || []) {
+                        // Remember to add the alias
+                        secureOptions.add(alias);
+                    }
+                } else if (prop.optionDefinitions) {
+                    // Handle the case of multiple option definitions
+                    prop.optionDefinitions.forEach(opDef => {
+                        secureOptions.add(opDef.name);
+                        for (const alias of opDef.aliases || []) {
+                            // Remember to add the alias
+                            secureOptions.add(alias);
+                        }
+                    });
+                } else {
+                    secureOptions.add(key);
+                }
+            }
+        }
+
+        secureOptions.forEach(prop => this.addCensoredOption(prop));
+    }
+
+    /**
+     * Add a censored option, including it's camelCase and kebabCase versions
+     * @param {string} option - The option to censor
+     */
+    private static addCensoredOption(option: string) {
+        // This option is required, but we do not want to ever allow null or undefined itself into the censored options
+        if (option != null) {
+            this.censored_options.add(option);
+            this.censored_options.add(CliUtils.getOptionFormat(option).camelCase);
+            this.censored_options.add(CliUtils.getOptionFormat(option).kebabCase);
+        }
+    }
+
+    /**
+     * Specifies whether a given property path (e.g. "profiles.lpar1.properties.host") is a special value or not.
+     * Special value: Refers to any value defined as secure in the schema definition.
+     *                These values should be already masked by the application (and/or plugin) developer.
+     * @param prop Property path to determine if it is a special value
+     * @returns True - if the given property is to be treated as a special value; False - otherwise
+     */
+    public static isSpecialValue(prop: string): boolean {
+        let specialValues = this.SECURE_PROMPT_OPTIONS;
+        const getPropertyNames = (prop: ICommandProfileProperty): string[] => {
+            const ret: string[] = [];
+            ret.push(prop.optionDefinition?.name);
+            prop.optionDefinitions?.map(opDef => ret.push(opDef.name));
+            return ret;
+        };
+
+        for (const profile of this.profileSchemas) {
+            // eslint-disable-next-line unused-imports/no-unused-vars
+            for (const [_, prop] of Object.entries(profile.schema.properties)) {
+                if (prop.secure) specialValues = lodash.union(specialValues, getPropertyNames(prop));
+            }
+        }
+
+        for (const v of specialValues) {
+            if (prop.endsWith(`.${v}`)) return true;
+        }
+        return false;
```

Comment:
Thoughts on joining these two possible return values? For example:

```suggestion
        return specialValues.some((v) => prop.endsWith(`.${v}`));
```